torch/csrc/autograd/VariableTypeUtils.h

```diff
@@ -444,6 +444,13 @@ inline std::vector<SavedVariable> make_saved_variable_list(
       });
 }
 
+inline std::vector<SavedVariable> make_saved_variable_list_foreach_output(
```

Comment:
nit: you could have added a boolean flag to the existing make_saved_variable_list function.
But sounds ok as well.